### PR TITLE
Implement Structure for BuilderGrid Component

### LINKED ISSUE #129: Implement BuilderGrid Structure

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR integrates the component responsible for displaying the builders related to a specific commit.

image

Closes #129

### CODE REVIEW

File: src/main/webapp/components/BuilderGrid.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import {getField} from "./utils/getField";
+
+/**
+ * Component used as an indicator for build status.
+ * Statuses are differentiated by alternative
+ * styling rules.
```
Comment (MagoDelBlocco) [inline src/main/webapp/components/BuilderGrid.js:7]: nit: please have one blank line between annotations and function description

**Review by MagoDelBlocco — APPROVED** [review]
LGTM

File: src/main/webapp/components/BuilderGrid.js (hunk 1 above)
Comment (samuelsogbesan) [reply src/main/webapp/components/BuilderGrid.js:7]: Resolved b5dd82a

File: src/main/webapp/components/BuilderGrid.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * @param {string} props.status the build status. Partially determines
+ * the styling of the rules.
+ */
+const StatusIndicator = (props) => {
+  const className = 'indicator #'.replace('#', props.status);
```
Comment (krasimir-google) [inline src/main/webapp/components/BuilderGrid.js:12]:
```suggestion
  const className = 'indicator '.concat(props.status);
```

File: src/main/webapp/components/BuilderGrid.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      {props.name}
+    </button>
+  );
+}
+
+/**
+ * Component used to display the list of builders that built a commit.
+ *
+ * @param {Object[]} props.builders the builders for the commit.
+ * @param {string} props.builders[].builder.status the build status of builder.
```
Comment (krasimir-google) [inline src/main/webapp/components/BuilderGrid.js:37]:
```suggestion
 * @param {string} props.builders[].status the build status of builder.
```
reading the code below it seems that each builder in the builders array directly contains the status and name field (similarly for the comment on the next line)

File: src/main/webapp/components/BuilderGrid.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    </button>
+  );
+}
+
+/**
+ * Component used to display the list of builders that built a commit.
+ *
+ * @param {Object[]} props.builders the builders for the commit.
+ * @param {string} props.builders[].status the build status of builder.
+ * @param {string} props.builders[].builder.name the name of the builder.
```
Comment (krasimir-google) [inline src/main/webapp/components/BuilderGrid.js:38]:
```suggestion
 * @param {string} props.builders[].name the name of the builder.
```
(similar to the previous comment)

**Review by krasimir-google — APPROVED** [review]
LGTM with one nit.

**Pull request merged**